### PR TITLE
Update compute to latest.

### PR DESCRIPTION
**Pull request by brendandburns**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@brendandburns__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (bburns). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

**Comment by cormacpayne** [comment]
@brendandburns Hey Brendan, could you update the generate.cmd file with the version of AutoRest and spec you are using?

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/ComputeManagementClient.cs [hunk 1]
```diff
@@ -108,9 +108,9 @@ public partial class ComputeManagementClient : Microsoft.Rest.ServiceClient<Comp
         public virtual IVirtualMachineScaleSetVMsOperations VirtualMachineScaleSetVMs { get; private set; }
 
         /// <summary>
-        /// Gets the IContainerServiceOperations.
+        /// Gets the IContainerServicesOperations.
         /// </summary>
-        public virtual IContainerServiceOperations ContainerService { get; private set; }
+        public virtual IContainerServicesOperations ContainerServices { get; private set; }
```
Comment (cormacpayne) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/ComputeManagementClient.cs]: @brendandburns where is the file for `IContainerServicesOperations`?

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/ContainerServiceOperations.cs [hunk 2]
```diff
@@ -1,873 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (cormacpayne) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/ContainerServiceOperations.cs]: @brendandburns I'm assuming that this file was deleted for `ContainerServicesOperations.cs`, but that file does not exist

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/ContainerServiceOperationsExtensions.cs [hunk 3]
```diff
@@ -1,271 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (cormacpayne) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/ContainerServiceOperationsExtensions.cs]: @brendandburns I'm assuming this file was deleted for `ContainerServicesOperationsExtensions.cs`, but that file does not exist

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/IComputeManagementClient.cs [hunk 4]
```diff
@@ -108,9 +108,9 @@ public partial interface IComputeManagementClient : System.IDisposable
         IVirtualMachineScaleSetVMsOperations VirtualMachineScaleSetVMs { get; }
 
         /// <summary>
-        /// Gets the IContainerServiceOperations.
+        /// Gets the IContainerServicesOperations.
         /// </summary>
-        IContainerServiceOperations ContainerService { get; }
+        IContainerServicesOperations ContainerServices { get; }
```
Comment (cormacpayne) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/IComputeManagementClient.cs]: @brendandburns same comment

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/IContainerServiceOperations.cs [hunk 5]
```diff
@@ -1,172 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (cormacpayne) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/IContainerServiceOperations.cs]: @brendandburns I'm assuming this file was deleted for `IContainerServicesOperations.cs`, but that file does not exist

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerService.cs [hunk 6]
```diff
@@ -24,26 +24,32 @@ public partial class ContainerService : Resource
...
+        /// <param name="windowsProfile">Properties of Windows VMs.</param>
+        /// <param name="diagnosticsProfile">Properties of the diagnostic
+        /// agent.</param>
+        public ContainerService(string location, ContainerServiceMasterProfile masterProfile, System.Collections.Generic.IList<ContainerServiceAgentPoolProfile> agentPoolProfiles, ContainerServiceLinuxProfile linuxProfile, string id = default(string), string name = default(string), string type = default(string), System.Collections.Generic.IDictionary<string, string> tags = default(System.Collections.Generic.IDictionary<string, string>), string provisioningState = default(string), ContainerServiceOrchestratorProfile orchestratorProfile = default(ContainerServiceOrchestratorProfile), ContainerServiceCustomProfile customProfile = default(ContainerServiceCustomProfile), ContainerServiceServicePrincipalProfile servicePrincipalProfile = default(ContainerServiceServicePrincipalProfile), ContainerServiceWindowsProfile windowsProfile = default(ContainerServiceWindowsProfile), ContainerServiceDiagnosticsProfile diagnosticsProfile = default(ContainerServiceDiagnosticsProfile))
```
Comment (cormacpayne) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerService.cs]: @brendandburns this is a breaking change - parameters have been added to the constructor and the order of some of the last parameters have changed

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceAgentPoolProfile.cs [hunk 7]
```diff
@@ -45,8 +44,10 @@ public partial class ContainerServiceAgentPoolProfile
         /// 'Standard_DS4', 'Standard_DS11', 'Standard_DS12',
         /// 'Standard_DS13', 'Standard_DS14', 'Standard_GS1', 'Standard_GS2',
         /// 'Standard_GS3', 'Standard_GS4', 'Standard_GS5'</param>
-        /// <param name="fqdn">FDQN for the agent pool</param>
-        public ContainerServiceAgentPoolProfile(string name, string dnsPrefix, int? count = default(int?), string vmSize = default(string), string fqdn = default(string))
+        /// <param name="dnsPrefix">DNS prefix to be used to create the FQDN
+        /// for the agent pool.</param>
+        /// <param name="fqdn">FDQN for the agent pool.</param>
+        public ContainerServiceAgentPoolProfile(string name, int count, string vmSize, string dnsPrefix, string fqdn = default(string))
```
Comment (cormacpayne) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceAgentPoolProfile.cs]: @brendandburns this is a breaking change - the order of the parameters in the constructor has changed and default values for parameters have been removed

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceAgentPoolProfile.cs [hunk 8]
```diff
@@ -56,17 +57,19 @@ public ContainerServiceAgentPoolProfile(string name, string dnsPrefix, int? coun
...
 
         /// <summary>
-        /// Gets or sets no. of agents (VMs) that will host docker containers
+        /// Gets or sets number of agents (VMs) to host docker containers.
+        /// Allowed values must be in the range of 1 to 100 (inclusive). The
+        /// default value is 1.
         /// </summary>
         [Newtonsoft.Json.JsonProperty(PropertyName = "count")]
-        public int? Count { get; set; }
+        public int Count { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceAgentPoolProfile.cs]: @brendandburns this is a breaking change - this property is no longer nullable

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceOrchestratorProfile.cs [hunk 9]
```diff
@@ -25,21 +25,31 @@ public partial class ContainerServiceOrchestratorProfile
...
         /// </summary>
-        /// <param name="orchestratorType">Specifies what orchestrator will be
-        /// used to manage container cluster resources. Possible values
-        /// include: 'Swarm', 'DCOS'</param>
-        public ContainerServiceOrchestratorProfile(ContainerServiceOchestratorTypes? orchestratorType = default(ContainerServiceOchestratorTypes?))
+        /// <param name="orchestratorType">The orchestrator to use to manage
+        /// container service cluster resources. Valid values are Swarm,
+        /// DCOS, and Custom. Possible values include: 'Swarm', 'DCOS',
+        /// 'Custom', 'Kubernetes'</param>
+        public ContainerServiceOrchestratorProfile(ContainerServiceOchestratorTypes orchestratorType)
```
Comment (cormacpayne) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceOrchestratorProfile.cs]: @brendandburns this is a breaking change - the default value for a parameter has been removed

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceOrchestratorProfile.cs [hunk 10]
```diff
@@ -25,21 +25,31 @@ public partial class ContainerServiceOrchestratorProfile
...
-        /// Gets or sets specifies what orchestrator will be used to manage
-        /// container cluster resources. Possible values include: 'Swarm',
-        /// 'DCOS'
+        /// Gets or sets the orchestrator to use to manage container service
+        /// cluster resources. Valid values are Swarm, DCOS, and Custom.
+        /// Possible values include: 'Swarm', 'DCOS', 'Custom', 'Kubernetes'
         /// </summary>
         [Newtonsoft.Json.JsonProperty(PropertyName = "orchestratorType")]
-        public ContainerServiceOchestratorTypes? OrchestratorType { get; set; }
+        public ContainerServiceOchestratorTypes OrchestratorType { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceOrchestratorProfile.cs]: @brendandburns same comment about changing nullable type

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceSshConfiguration.cs [hunk 11]
```diff
@@ -25,19 +25,42 @@ public partial class ContainerServiceSshConfiguration
         /// Initializes a new instance of the ContainerServiceSshConfiguration
         /// class.
         /// </summary>
-        /// <param name="publicKeys">the list of SSH public keys used to
-        /// authenticate with Linux based VMs</param>
-        public ContainerServiceSshConfiguration(System.Collections.Generic.IList<ContainerServiceSshPublicKey> publicKeys = default(System.Collections.Generic.IList<ContainerServiceSshPublicKey>))
+        /// <param name="publicKeys">The list of SSH public keys used to
+        /// authenticate with Linux-based VMs.</param>
+        public ContainerServiceSshConfiguration(System.Collections.Generic.IList<ContainerServiceSshPublicKey> publicKeys)
```
Comment (cormacpayne) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceSshConfiguration.cs]: @brendandburns same comment about removing default value

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceVMDiagnostics.cs [hunk 12]
```diff
@@ -25,29 +25,38 @@ public partial class ContainerServiceVMDiagnostics
...
-        /// <param name="enabled">whether VM Diagnostic Agent should be
-        /// provisioned on the Virtual Machine.</param>
-        /// <param name="storageUri">whether VM Diagnostic Agent should be
-        /// provisioned on the Virtual Machine.</param>
-        public ContainerServiceVMDiagnostics(bool? enabled = default(bool?), string storageUri = default(string))
+        /// <param name="enabled">Whether the VM diagnostic agent is
+        /// provisioned on the VM.</param>
+        /// <param name="storageUri">The URI of the storage account where
+        /// diagnostics are stored.</param>
+        public ContainerServiceVMDiagnostics(bool enabled, string storageUri = default(string))
```
Comment (cormacpayne) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceVMDiagnostics.cs]: @brendandburns same comment about removing default value

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceVMDiagnostics.cs [hunk 13]
```diff
@@ -25,29 +25,38 @@ public partial class ContainerServiceVMDiagnostics
...
 
         /// <summary>
-        /// Gets or sets whether VM Diagnostic Agent should be provisioned on
-        /// the Virtual Machine.
+        /// Gets or sets whether the VM diagnostic agent is provisioned on the
+        /// VM.
         /// </summary>
         [Newtonsoft.Json.JsonProperty(PropertyName = "enabled")]
-        public bool? Enabled { get; set; }
+        public bool Enabled { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/Generated/Models/ContainerServiceVMDiagnostics.cs]: @brendandburns same comment about changing nullable type

**Review by cormacpayne — CHANGES_REQUESTED** [review]
@brendandburns left a few comments that will need to be addressed

You will also need to update the `project.json` and `AssemblyInfo.cs` files with the appropriate version bump for these changes

**Comment by brendandburns** [comment]
@cormacpayne thanks for the feedback.  Added the missing files (I forgot to `git add ...` *sigh*)

What is the policy on breaking changes?

**Comment by shahabhijeet** [comment]
@brendandburns please address the feedback on the PR. We will close the PR by EOW. You can either open a new PR or reopen the existing one once the feedback has been addressed.

**Comment by cormacpayne** [comment]
@brendandburns Hey Brendan, since there hasn't been any activity in this PR in a couple of weeks, I'm going to close it. Feel free to re-open it when comments can be addressed.

**Comment by brendandburns** [comment]
Re-opening

@cormacpayne I think this is ready to go since I re-added the files. Are there any other comments you need me to address?

**Comment by azurecla** [comment]
Hi __@brendandburns__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (bburns). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by brendandburns** [comment]
Looks like I need to update tests, I will address later to day.

**Comment by markcowl** [comment]
@brendandburns Two VM tests are failg in this check-in, please take a look

**Comment by markcowl** [comment]
It has been 19 days since this PR was changed.  Closing for now Please open a new PR when you are ready to proceed.

**Pull request closed without merging**